tools/refract-filter.py

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python
+
+from __future__ import print_function
+import argparse
+import sys
+import yaml
+import json
+import textwrap
+from collections import OrderedDict
+
+try:
+    from StringIO import StringIO
+except ImportError:
+    from io import StringIO
+
+VERSION = "0.1"
+
+def yaml_load(stream, Loader=yaml.Loader, object_pairs_hook=OrderedDict):
+    class OrderedLoader(Loader):
+        pass
+    def construct_mapping(loader, node):
+        loader.flatten_mapping(node)
+        return object_pairs_hook(loader.construct_pairs(node))
+
+    OrderedLoader.add_constructor(
+        yaml.resolver.BaseResolver.DEFAULT_MAPPING_TAG,
+        construct_mapping)
+    return yaml.load(stream, OrderedLoader)
+
+def walk(node, cb):
+
+    if type(node) is OrderedDict:
```

Comment:
What about doing:

```python
if isinstance(node, dict):
```

So these statements are less coupled to the implementation details of what type of dictionary these are (ordered, non-ordered).